src/vorta/keyring/secretstorage.py

```diff
@@ -65,6 +65,24 @@ def get_password(self, service, repo_url):
                 return item.get_secret().decode("utf-8")
         return None
 
+    def remove_password(self, service, repo_url):
+        """
+        Remove a password from the underlying store.
+        """
+        if self.is_unlocked:
+            asyncio.set_event_loop(asyncio.new_event_loop())
+            attributes = {
+                'application': 'Vorta',
+                'service': service,
+                'repo_url': repo_url,
+            }
+            items = list(self.collection.search_items(attributes))
+            logger.debug('Found %i passwords matching repo URL.', len(items))
+            for item in items:
+                self.collection.delete_item(item)
```

Comment:
Does this method even exist? Secondly, i think we also have to check whether the item is unlocked before we can delete it.

Reply:
My bad its not *delete_item* its just [*delete()*](https://secretstorage.readthedocs.io/en/latest/_modules/secretstorage/collection.html#Collection.delete) . We have already checked if the item is unlocked in `if self.is_unlocked:`

Reply:
Nope, `is_unlocked` only checks whether the collection is unlocked while it is possible with some key managers that the items are locked individually as well. This is why we added a second test in `get_password`.

Reply:
So basically i have to check `if self.is_unlocked` again in the for loop ?

Reply:
See
https://github.com/borgbase/vorta/blob/abb3ec82ee1de301355965e4d46e21c4e69064d2/src/vorta/keyring/secretstorage.py#L62-L63